### PR TITLE
OIDC auth

### PR DESCRIPTION
**Pull request by tmulle**
Added automatic refresh/logout dialog to the demo using custom javascript and logout dialog xhtml

### CODE REVIEW

**Comment by melloware** [comment]
Looks like there is some merge conflicts?

**Comment by tmulle** [comment]
I clicked on the "Resolve Conflicts" button but there doesn't appear to be a way for me to fix them. I guess I don't have write permissions. I can only view the conflicts.

image

Not sure why it's conflicting, I just added methods and comments

**Comment by melloware** [comment]
Looks like you didn't update your fork branch after I merged.

You should always get from master and then create a new branch.

**Comment by melloware** [comment]
OK i just tried to resolve them can you check them?

**Comment by melloware** [comment]
actually yeah its showing 335 file still.

**Comment by melloware** [comment]
you should get from `master` then create a new branch from that. I should only be seeing your changes here but it looks like you just re-committed the old branch.

File: src/main/java/org/primefaces/showcase/service/SecurityService.java [hunk 1]
```diff
@@ -22,7 +27,78 @@ public class SecurityService {
     @Inject
     SecurityIdentity identity;
 
- 
+    @ConfigProperty(name = "ssoIdleTimeout", defaultValue = "PT30M")
```
Comment (melloware) [inline src/main/java/org/primefaces/showcase/service/SecurityService.java:30]:
```suggestion
    @ConfigProperty(name = "sso.idle.timeout", defaultValue = "PT30M")
```

File: src/main/java/org/primefaces/showcase/service/SecurityService.java [hunk 2]
```diff
@@ -22,7 +27,78 @@ public class SecurityService {
...
+    Duration idleTimeout;
+
+    @ConfigProperty(name = "ssoIdelTimeoutWarning", defaultValue = "PT1M")
```
Comment (melloware) [inline src/main/java/org/primefaces/showcase/service/SecurityService.java:33]:
```suggestion
    @ConfigProperty(name = "sso.idle.timeout.warning", defaultValue = "PT1M")
```

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -69,5 +68,27 @@ quarkus.security.jaxrs.deny-unannotated-endpoints=false
...
+# In the real world this would match Keycloak setting
+ssoIdleTimeout=PT30S
+
+# This is how soon before the actual timeout should the 
+# warning dialog be shown
+# Usually 1 minute before the SSO Idle timeout is enough
+# 
+# I'm setting 10s countdown for the demo
+# Make it longer in the real world
+ssoIdelTimeoutWarning=PT10S
```
Comment (melloware) [inline src/main/resources/application.properties:92]: Fix these

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/java/org/primefaces/showcase/service/SecurityService.java (hunk 1 above)
Comment (tmulle) [reply src/main/java/org/primefaces/showcase/service/SecurityService.java]: Yup..made the change

File: src/main/java/org/primefaces/showcase/service/SecurityService.java (hunk 2 above)
Comment (tmulle) [reply src/main/java/org/primefaces/showcase/service/SecurityService.java]: Made this one too

File: src/main/resources/application.properties (hunk 3 above)
Comment (tmulle) [reply src/main/resources/application.properties]: changed the naming

**Review by melloware — APPROVED** [review]

**Pull request merged**